<!-- RoboPuppet Arm Launch File -->
<!-- Launches RoboPuppet nodes for L or R side -->
<launch>
	
	<!-- Arguments -->
	<arg name="arm_side" default="L" doc="Arm side [L, R]"/>
	<arg name="port_name" default="/dev/ttyACM0" doc="Serial port name"/>
	<arg name="comm_rate" default="10.0" doc="Communication rate [Hz]"/>
	<arg name="frame_rate" default="8.0" doc="GUI framerate [Hz]"/>
	<arg name="debug_mode" default="true" doc="Debug mode flag [Bool]"/>
	
	<!-- Config File Manager -->
	<node pkg="robopuppet" type="config.py" name="config_$(arg arm_side)" output="screen">
		<param name="arm_side" type="str" value="$(arg arm_side)"/>
		<param name="file_name" type="str" value="$(find robopuppet)/config/arm_$(arg arm_side).ini"/>
	</node>

	<!-- Serial Interface -->
	<node pkg="robopuppet" type="serial_interface.py" name="serial_$(arg arm_side)" output="screen">
		<param name="arm_side" type="str" value="$(arg arm_side)"/>
		<param name="port_name" type="str" value="$(arg port_name)"/>
		<param name="baud_rate" type="int" value="115200"/>
		<param name="comm_rate" type="double" value="$(arg comm_rate)"/>
	</node>
	
	<!-- Gripper Controller -->
	<node pkg="robopuppet" type="gripper.py" name="gripper_$(arg arm_side)" output="screen">
		<param name="arm_side" type="str" value="$(arg arm_side)"/>
		<param name="comm_rate" type="double" value="$(arg comm_rate)"/>
	</node>
	
	<!-- Arm Controller -->
	<node pkg="robopuppet" type="arm.py" name="arm_$(arg arm_side)" output="screen">
		<param name="arm_side" type="str" value="$(arg arm_side)"/>
		<param name="comm_rate" type="double" value="$(arg comm_rate)"/>
	</node>

	<!-- Control GUI -->
	<node pkg="robopuppet" type="gui.py" name="gui_$(arg arm_side)" output="screen">
		<param name="arm_side" type="str" value="$(arg arm_side)"/>
		<param name="frame_rate" type="double" value="$(arg frame_rate)"/>
		<param name="debug_mode" type="bool" value="$(arg debug_mode)"/>
	</node>
</launch>
